core/net/mac/tsch/sixtop/sixp.c

```diff
@@ -0,0 +1,967 @@
+} sixp_trans_t;
+
+/**
+ * \brief 6P Neighbor Data Structure (for internal use)
+ *
+ * XXX: for now, we have one nbr object per neighbor, which is shared with
+ * multiple SFs. It's unclear whether we should use a different generation
+ * counter for each SFs.
+ */
+typedef struct sixp_nbr {
+  struct sixp_nbr *next;
+  linkaddr_t addr;
+  uint8_t next_seqno;
+  uint8_t gtx;
+  uint8_t grx;
+} sixp_nbr_t;
+
+static void
+print_msg_event_log(sixp_type_t type, sixp_code_t code,
+                    const linkaddr_t *peer_addr, sixp_log_event_t event);
+
+static sixp_nbr_t *find_nbr(const linkaddr_t *addr);
+static sixp_nbr_t *alloc_nbr(const linkaddr_t *addr);
+#if 0
+static void free_nbr(sixp_nbr_t *nbr);
+static void clear_all_nbr(void);
+#endif
+
+static void handle_trans_timeout(void *ptr);
+static void process_trans(void *ptr);
+static void schedule_trans_process(sixp_trans_t *trans);
+static int transit_trans_state(sixp_trans_t *trans,
+                               sixp_trans_state_t new_state);
+static sixp_trans_t *alloc_trans(sixp_code_t code, uint8_t sfid, uint8_t seqno,
+                                 const linkaddr_t *peer_addr,
+                                 sixp_trans_mode_t mode);
+static void free_trans(sixp_trans_t *trans);
+static sixp_trans_t *find_trans(const linkaddr_t *peer_addr);
+static int trans_state_is_invalid(sixp_type_t sending_msg_type,
+                                  sixp_code_t code,
+                                  const linkaddr_t *peer_addr);
+static sixp_trans_mode_t determine_trans_mode(sixp_command_id_t cmd,
+                                              uint16_t body_len);
+
+static int schedule_generation_is_invalid(const sixtop_ie_t *sixtop_ie,
+                                          const linkaddr_t *peer_addr);
+
+static int prepare_request(sixp_code_t code, uint8_t sfid,
+                           const linkaddr_t *peer_addr, sixp_trans_mode_t mode);
+static int fill_sixtop_ie(sixp_type_t type, sixp_code_t code, uint8_t sfid,
+                          uint8_t seqno,
+                          const sixtop_ie_body_t *body, uint16_t body_len,
+                          const linkaddr_t *peer_addr,
+                          sixtop_ie_t *ie);
+static void mac_callback(void *ptr, int status, int transmissions);
+static int send_6p_packet(sixp_type_t type, sixp_code_t code, uint8_t sfid,
+                          uint8_t seqno,
+                          const sixtop_ie_body_t *body, uint16_t body_len,
+                          const linkaddr_t *dest_addr);
+
+static void request_input(const sixtop_ie_t *ie, const linkaddr_t *peer_addr);
+static void response_input(const sixtop_ie_t *ie, const linkaddr_t *peer_addr);
+static void confirmation_input(const sixtop_ie_t *ie,
+                               const linkaddr_t *peer_addr);
+
+
+MEMB(trans_memb, sixp_trans_t, SIXTOP_6P_MAX_TRANSACTIONS);
+LIST(trans_list);
+
+MEMB(nbr_memb, sixp_nbr_t, SIXTOP_6P_MAX_NEIGHBORS);
```

Comment:
I think so too, but please note that nbr-table is to be used only outside of interrupt (read and write). I think that should be fine with 6p but just checking.